File: Java-Module/src/main/java/com/sim/javamodule/domain/user/domain/CreditRepository.java
package com.sim.javamodule.domain.user.domain;

import jakarta.persistence.LockModeType;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Lock;

import java.util.Optional;

public interface CreditRepository extends JpaRepository<Credit, Long> {

    @Lock(value = LockModeType.PESSIMISTIC_WRITE)
    Optional<Credit> findByUserId(Long userId);
}

File: Kotlin-Module/src/main/kotlin/com/sim/kotlin/domain/user/domain/CreditRepository.kt
package com.sim.kotlin.domain.user.domain

import jakarta.persistence.LockModeType
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.Lock
import java.util.Optional

interface CreditRepository:JpaRepository<Credit, Long> {

    @Lock(value = LockModeType.PESSIMISTIC_WRITE)
    fun findByUserId(userId: Long) : Credit?
}